provider/agent_test.go

```diff
@@ -247,3 +248,106 @@ func TestAgent_Metadata(t *testing.T) {
 		}},
 	})
 }
+
+func TestAgent_DefaultApps(t *testing.T) {
+	t.Parallel()
+	t.Run("OK", func(t *testing.T) {
+		resource.Test(t, resource.TestCase{
+			Providers: map[string]*schema.Provider{
+				"coder": provider.New(),
+			},
+			IsUnitTest: true,
+			Steps: []resource.TestStep{{
+				// Test the fields with non-default values.
+				Config: `
+					provider "coder" {
+						url = "https://example.com"
+					}
+					resource "coder_agent" "dev" {
+						os = "linux"
+						arch = "amd64"
+						display_apps {
+							vscode = false
```

Comment:
Do we need an extra test case to check .tf with only a few apps selected:

```
display_apps {
   vscode = true
   web_terminal = true
```

Reply:
Since they all have default values I tested that specifying their non-defaults (in `OK`) and then omitting them to produce the defaults (in `Omitted`) produces the desired result. I added another one anyway but I don't think it adds much value.